plover/command/set_config.py

```diff
@@ -0,0 +1,29 @@
+from plover import log
+
+
+def set_config(engine, cmdline):
+    """
+    Set a Plover config option to a given value upon executing a stroke pattern.
+
+    Example usage (enter these lines into dictionary):
+    "O*EP": "{PLOVER:SET_CONFIG:translation_frame_opacity,100}",
+    "TR*PB": "{PLOVER:SET_CONFIG:translation_frame_opacity,0}",
+    """
+    # Break down the command string into a key and a value. Throw an error if it isn't in the form of a 2-tuple.
+    try:
+        key, value = tuple(cmdline.split(','))
+    except ValueError:
+        log.error("SET_CONFIG: Bad command string. Command must be in the form {PLOVER:SET_CONFIG:option,value}")
+        return
+    # We need to look up the old value of the config setting in order to determine its type.
+    try:
+        oldvalue = engine[key]
+    except KeyError:
+        log.error("SET_CONFIG: %s is not a valid Plover config option." % key)
+        return
+    # Type-cast the value string to match whatever type the old value was, and set the option to the new value.
+    try:
+        engine[key] = (type(oldvalue))(value)
+    except (TypeError, ValueError):
+        log.error("SET_CONFIG: Could not cast config value string \"%s\" to %s." % (value, type(oldvalue)))
+        return
```

Comment:
Same here for the error handling.

Reply:
I don't like the way de-serializing from a string is handled, I think a more robust approach would be to use `ast.literal_eval`:
```python
def set_config(engine, cmdline):
    """
    Set a Plover config option to a given value upon executing a stroke pattern.

    Example usage (enter these lines into dictionary):
    "O*EP": "{PLOVER:SET_CONFIG:translation_frame_opacity:100}",
    "TR*PB": "{PLOVER:SET_CONFIG:translation_frame_opacity:0}",
    """
    key, value = tuple(cmdline.split(':', 1))
    engine[key] = ast.literal_eval(value)
```
This would make it possible to use the command with complex settings, like machine options: `{PLOVER:SET_CONFIG:machine_specific_options:\{'arpeggiate': True\}}`.

Reply:
Aha, that's perfect! I was looking for a way to handle the more complex config options! Thanks!